packages/dapp/src/components/Parameterizer.tsx

```diff
@@ -110,19 +110,15 @@ class Parameterizer extends React.Component<{}, ParameterStates> {
   }
 
   private initParameterValues = async (): Promise<any> => {
-    const parameterObj = {};
     const keys = Object.keys(this.state);
-    let i = 0;
 
-    keys.forEach(async parameterKey => {
-      const parameterVal = await getParameterValue(parameterKey);
-      parameterObj[parameterKey] = parameterVal.toString();
-      i++;
+    const parameterVals = await getParameterValue(keys);
```

Comment:
So fresh and so clean! 

So I created a reducer for the parameterizer state in another PR, but it's based on the old `initParamValues`. Would love to update my PR with this once yours lands.